backend/db/resolve_fetch_failures_0_members.py

```diff
@@ -10,35 +11,87 @@
 from typing import Dict, List, Set, Tuple, Union
 
+from sqlalchemy import Connection
+
 DB_DIR = os.path.dirname(os.path.realpath(__file__))
 BACKEND_DIR = os.path.join(DB_DIR, "..")
 PROJECT_ROOT = os.path.join(BACKEND_DIR, "..")
 sys.path.insert(0, str(PROJECT_ROOT))
 from backend.utils import call_github_action
 from backend.db.resolve_fetch_failures_excess_items import resolve_fetch_failures_excess_items
 from backend.db.utils import SCHEMA, fetch_status_set_success, get_db_connection, reset_temp_refresh_tables, \
-    select_failed_fetches, refresh_derived_tables, sql_in, sql_query
+    run_sql, select_failed_fetches, refresh_derived_tables, sql_in, sql_query
 from enclave_wrangler.objects_api import concept_set_members__from_csets_and_members_to_db, \
-    fetch_cset_and_member_objects, get_csets_over_threshold
+    fetch_cset_and_member_objects, fetch_cset_version, get_csets_over_threshold
 
 DESC = "Resolve any failures resulting from fetching data from the Enclave's objects API."
 
 
 def _report_success(
-    cset_ids: List[int], fetch_audit_row_lookup: Dict[int, Dict], comment_addition: str = None, use_local_db=False
+    cset_ids: Union[int, List[int]], fetch_audit_row_lookup: Dict[int, Dict], comment_addition: str = None, use_local_db=False
 ):
-    """Report success for a list of concept set IDs.
+    """Report success fetch success for given concept sets.
     todo: kind of weird to pass a lookup. Maybe do that beforehand and pass row dict"""
+    cset_ids = [cset_ids] if not isinstance(cset_ids, list) else cset_ids
     success_rows = []
     for cset_id in cset_ids:
         row = fetch_audit_row_lookup[cset_id]
         if comment_addition:
             row['comment'] = row['comment'] + '; ' + \
-            'Success result: ' if not comment_addition.startswith('Success result: ') else '' + comment_addition
+                 ('Success result: ' if not comment_addition.startswith('Success result: ') else '') + comment_addition
         success_rows.append(row)
     fetch_status_set_success(success_rows, use_local_db)
 
 
+def _ad_hoc_detect_and_delete_discarded_drafts(schema: str = SCHEMA, local: bool = False):
+    """Check all csets currently marked draft in the DB, check if discarded, and handle."""
+    with get_db_connection(schema=schema, local=local) as con:
+        rows: List[List[int]] = sql_query(
+            con, f"SELECT codeset_id FROM code_sets WHERE is_draft = true;", return_with_keys=False)
+        draft_ids: List[int] = [x[0] for x in rows]
+        cset_versions: List[Dict] = [fetch_cset_version(_id, retain_properties_nesting=True) for _id in draft_ids]
+        cset_versions = [x for x in cset_versions if x]  # remove discarded draft empty dicts {}
+        csets_and_members: Dict[str, List[Dict]] = {'OMOPConceptSet': cset_versions}
+        handle_discarded_drafts(con, csets_and_members, set(draft_ids), report_success=False, use_local_db=local)
+
+
+def handle_discarded_drafts(
```

Comment:
## `handle_discarded_drafts()`
Now its own function, so it can be run ad hoc, independent of fetch failures.

Now no longer just marks success for fetch failures in `fetch_audit` when drafts have been discarded, but also deletes from the DB.

It also saves a backup of the properties of the cset (from the `code_sets` table) as a JSON string that it puts that in the `comment` field in `fetch_audit`.

I just now realized there are two potential improvements here:
1. I could have serialized the row from `all_csts` instead of `code_sets`, for more information.
2. If running this ad hoc, no backup is created (I ran this 1x and created it manually). Should do this programmatically; could just save to JSON at the root of TermHub.